src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -1476,12 +1476,39 @@ private void doGenCert(String alias, String sigAlgName, InputStream in, PrintStr
                 reqex = (CertificateExtensions)attr.getAttributeValue();
             }
         }
+
+        PublicKey subjectPubKey = req.getSubjectPublicKeyInfo();
+        PublicKey issuerPubKey = signerCert.getPublicKey();
+
+        KeyIdentifier signerSubjectKeyId;
+        if (subjectPubKey.equals(issuerPubKey)) {
```

Comment:
I think in most cases, this equality test will not work as there is no requirement for PublicKey to override Object.equals, so in most cases this will just check if they reference the same object. I suggest comparing the encoded bytes.

Reply:
Original logic using this equality test. Fixed as suggested.